linkcheck/logger/__init__.py

```diff
@@ -130,7 +130,7 @@ def log_internal_error (self):
         self.internal_errors += 1
 
 
-class _Logger (object):
+class _Logger (abc.ABC):
```

Comment:
Where do you stand on the space between the class name and the bracket? We have both. I have got another commit removing all of them - I mistakenly thought they caused flake8 E211 errors. As they don't I could junk that commit.

Reply:
> Where do you stand on the space between the class name and the bracket? We have both. I have got another commit removing all of them - I mistakenly thought they caused flake8 E211 errors. As they don't I could junk that commit.

Ah! Whitespace! *FINALLY* something I can have a strong opinion and definite skill on! ;)

Joking aside, this is one thing PEP8 is (ironically) silent on. it doesn't explicitely mandate that, strangely enough. flake8 also does not warn on this, so if we pick a policy on this, we might have trouble applying it in the future.

That said, pep8 does say that we should avoid spaces after *functions*:

https://www.python.org/dev/peps/pep-0008/#whitespace-in-expressions-and-statements

... so I would argue a class declaration should follow that model.

It's also what [Black](https://github.com/psf/black) does, so I would remove the space, which looks odd to me.

This, obviously, is a rather trivial matter that we shouldn't bother with too much. ;) The first rule of PEP8 is, of course:

> A Foolish Consistency is the Hobgoblin of Little Minds

Reply:
I hate these spaces.  I've always thought that PEP-8 forbids them as well.  Sometimes I'm tempted to run [`black`](https://pypi.org/project/black/) on the source tree, but I don't because that might make life harder for any currently outstanding PRs.

As usual, consistency with existing code trumps PEP-8 (until we collectively decide to reformat all the existing code), and so I haven't been changing the style of existing code.